frontend/Interfaces/buddy/DIP/ImageContainer.h

```diff
@@ -22,24 +22,254 @@
 #include <cassert>
 #include <opencv2/opencv.hpp>
 
+using namespace dip;
 // Image container.
 // - T represents the type of the elements.
 // - N represents the number of dimensions.
-// - image represents the OpenCV Mat object.
-// - norm indicates whether to perform normalization, and the normalization is
-//   disabled by default.
 template <typename T, size_t N> class Img : public MemRef<T, N> {
 public:
+  Img();
+
+  /**
+   * @brief overload
+   * @param rows Number of rows in a 2D array.
+   * @param cols Number of columns in a 2D array.
+   * @param type Array type. Use IMG_8UC1, ..., IMG_64FC4 to create 1-4 channel
+   * matrices.
+   */
+  Img(int rows, int cols, int type);
+
+  /**
+   * @brief overload
+   * @param rows Number of rows in a 2D array.
+   * @param cols Number of columns in a 2D array.
+   * @param type Array type. Use IMG_8UC1, ..., IMG_64FC4 to create 1-4 channel
+   * matrices.
+   * @param data Pointer to the user data.
+   * they just initialize the matrix header that points to the specified data.
+   */
+  Img(int rows, int cols, int type, T *data);
+
+  /**
+   * @brief overload
+   * @param ndims Array dimensionality.
+   * @param sizes Array of integers specifying an n-dimensional array shape.
+   * @param type Use IMG_8UC1, ..., IMG_64FC4 to create 1-4 channel
+   * matrices.
+   */
+  Img(int ndims, intptr_t *sizes, int type);
+
+  /**
+   * @brief overload
+   * @param m Array that (as a whole or partly) is assigned to the constructed
+   * matrix.
+   */
+  Img(const Img<T, N> &m);
+
+  /**
+   * @brief assignment operators
+   * @param m Assigned, right-hand-side matrix.
+   * matrix.
+   */
+  Img &operator=(const Img<T, N> &m);
+
+  /**
+   * @brief Load image data from OpenCV Mat.
+   * @param image represents the OpenCV Mat object.
+   * @param norm indicates whether to perform.
+   */
   Img(cv::Mat image, intptr_t sizes[N] = nullptr, bool norm = false);
 
-private:
-  // Load image data from OpenCV Mat.
-  void loadImg(cv::Mat image, bool norm);
+  /**
+   * @brief Allocates new array data if needed.
+   * @param rows Number of rows in a 2D array.
+   * @param cols Number of columns in a 2D array.
+   * @param type Array type. Use IMG_8UC1, ..., IMG_64FC4 to create 1-4 channel
+   * matrices.
+   */
+  void create(int rows, int cols, int type);
+
+  /**
+   * @brief overload
+   * @param ndims Array dimensionality.
+   * @param sizes Array of integers specifying an n-dimensional array shape.
+   * @param type Use IMG_8UC1, ..., IMG_64FC4 to create 1-4 channel
+   * matrices.
+   */
+  void create(int ndims, intptr_t *sizes, int type);
+
+  int channels() const;
+
+  int _cols() const;
+
+  int _rows() const;
+
+  size_t total();
+
+  int flags;
+
+  // the matrix dimensionality, >= 2
+  int dims;
+
+  // the number of rows and columns or (-1, -1) when the matrix has more than 2
+  // dimensions
+  int rows, cols;
+
+  // Use IMG_8UC1, ..., IMG_64FC4 to create 1-4 channel matrices.
+  int _type;
```

Comment:
Remove '_' before variable name(function name as well). If you want to make it private, please use 'private' keyword.